explorer/interpreter/interpreter.cpp

```diff
@@ -423,30 +424,32 @@ auto Interpreter::StepLvalue() -> ErrorOr<Success> {
         //    { {e.f :: C, E, F} :: S, H}
         // -> { e :: [].f :: C, E, F} :: S, H}
         return todo_.Spawn(std::make_unique<LValAction>(&access.object()));
-      } else {
-        if (auto constant_value = access.constant_value()) {
-          CARBON_ASSIGN_OR_RETURN(
-              Nonnull<const Value*> instantiated,
-              InstantiateType(*constant_value, access.source_loc()));
-          return todo_.FinishAction(instantiated);
+      } else if (act.pos() == 1) {
+        auto constant_value = access.constant_value();
+        if (constant_value) {
+          return todo_.Spawn(std::make_unique<TypeInstantiationAction>(
+              *constant_value, access.source_loc()));
         }
```

Comment:
I think 
```
} else {
  ...
  return ...
}
```
might be easier to parse here, otherwise it looks like we do 2 `FinishAction` one after the other at a glance (even if that's not the case)

Reply:
Can I write it like this? but here I didn't use if with initializer for `constant_value` as it is used at two places.
```cpp
    case ExpressionKind::SimpleMemberAccessExpression: {
      const auto& access = cast<SimpleMemberAccessExpression>(exp);
      const auto constant_value = access.constant_value();
      if (auto rewrite = access.rewritten_form()) {
        return todo_.ReplaceWith(std::make_unique<LValAction>(*rewrite));
      }
      if (act.pos() == 0) {
        //    { {e.f :: C, E, F} :: S, H}
        // -> { e :: [].f :: C, E, F} :: S, H}
        return todo_.Spawn(std::make_unique<LValAction>(&access.object()));
      } else if (act.pos() == 1 && constant_value) {
          return todo_.Spawn(std::make_unique<TypeInstantiationAction>(
              *constant_value, access.source_loc()));
      } else {
        if(constant_value) {
          return todo_.FinishAction(act.results().back());
        } else {
          //    { v :: [].f :: C, E, F} :: S, H}
          // -> { { &v.f :: C, E, F} :: S, H }
          Address object = cast<LValue>(*act.results()[0]).address();
          Address member = object.ElementAddress(&access.member());
          return todo_.FinishAction(arena_->New<LValue>(member));
        }
      }
    }
    ```

Reply:
It tends to be easier to understand when the conditions reflect `act.pos()`, so at a high level you can see 
* Pos1
* Pos2
* ...

If there were more code reuse maybe it would make more sense, but here I think I would still recommend branching by `act.pos()` only.

Reply:
When branching with just `act.pos()`, I'm facing a little issue here.
Look that `What should I return here?` comment in the code below. If I don't return anything there will will fall through in the switch statement. How should I handle this case?
```cpp
if (act.pos() == 0) {
        ...
} else if (act.pos() == 1 && ) {
    if(constant_value) {
        return todo_.Spawn(std::make_unique<TypeInstantiationAction>(
           *constant_value, access.source_loc()));
    } else {
       // What should I return here?
    }
} else {
       ...
}
```

Reply:
Oh, I got it!! Thanks.